source/common/upstream/outlier_detection_impl.cc

```diff
@@ -609,6 +638,63 @@ void DetectorImpl::processSuccessRateEjections(
   }
 }
 
+void DetectorImpl::processFailurePercentageEjections(
```

Comment:
can we refactor processSuccessRateEjections for shared utilities?  It seems like there's a lot of overlap

Reply:
I took a first pass at this - there's still an unfortunate amount of duplication within the function due to the interactions of minimum hosts/request volume, but I was able to drop a few lines. Do you see any other obvious areas to improve?